src/net/larse/lcms/algorithms/VCT.java

```diff
@@ -516,14 +409,18 @@ private void interpolationAndIndices() {
           // No acceptable previous QA_GOOD - use next index to fill
           // all years from 0 to next
           else if (prev < 0) {
-            fillValues(0, next, next);
-            i = next + 1;
+            for (int k = 0; k < N_BANDS; k++) {
+              Arrays.fill(this.ud[k], 0, next, this.ud[k][next]);
+              // fillValues(0, next, next);
+            }
           }
           // No acceptable next QA_GOOD - use prev index to fill
           // all years from prev + 1 to num_years
           else if (next >= this.numYears) {
-            fillValues(prev+1, this.numYears, prev);
-            i = next + 1;
+            for (int k = 0; k < N_BANDS; k++) {
+              Arrays.fill(this.ud[k], prev + 1, this.numYears, this.ud[k][prev]);
+              // fillValues(prev+1, this.numYears, prev);
```

Comment:
I opted not to use the fillValues function as I think the for loop with the Arrays.fill is clear enough.  But overrule me if you want.